app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -1,17 +1,62 @@
 package com.jetbrains.handson.mpp.mobile
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.Intent
+import android.net.Uri
 import android.os.Bundle
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import android.widget.Button
+import android.widget.Spinner
 import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+
 
 class MainActivity : AppCompatActivity(), ApplicationContract.View {
 
+    private fun initStationSpinner(spinnerId: Int, listOfStations: List<Station>, setter: (Station) -> Unit) {
+        val spinner: Spinner = findViewById(spinnerId) // Create an ArrayAdapter using the string array and a default spinner layout.
+
+        val adapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, listOfStations.map { it.getName() })
+
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+
+        spinner.adapter = adapter
+
+        spinner.onItemSelectedListener = object: AdapterView.OnItemSelectedListener{
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
```

Comment:
This is a blank method doing nothing, do you need it?

Reply:
We think so; we tried removing it, and it complained that we didn't implement the abstract onNothingSelected() function declared in AdapterView.OnItemSelectedListener

Reply:
Did it fail to compile? It feels like a nasty case of YAGNI, but if it's forced then we'll live with it!